pkg/pillar/zedcloud/tls.go

```diff
@@ -172,6 +172,30 @@ func GetTlsConfig(dns *types.DeviceNetworkStatus, serverName string, clientCert
 	return tlsConfig, nil
 }
 
+// UpdateTLSProxyCerts - Update when DeviceNetworkStatus changes
+func UpdateTLSProxyCerts(ctx *ZedCloudContext) {
+	tlsCfg := ctx.TlsConfig
+	devNS := ctx.DeviceNetworkStatus
+	if tlsCfg == nil || devNS == nil {
+		log.Errorln("UpdateTLSProxyCerts: tlsconfig or dev NS missing")
+		return
+	}
+
+	var updated bool
+	caCertPool := tlsCfg.RootCAs
+	// AppendCertsFromPEM checks duplicates inside
```

Comment:
Can it return something based on duplicates? Or is there a function to ask if a cert is already present in the pool?

Reply:
I checked, there is no directly function we can call to check on duplicates. the internal function s.AddCert(cert), first checks this if s.contains(cert) is true, then immediately return.